cmd/troubleshoot/cli/run.go

```diff
@@ -327,14 +328,22 @@ func loadSpecs(ctx context.Context, args []string, client kubernetes.Interface)
 			APIVersion: "troubleshoot.sh/v1beta2",
 			Kind:       "SupportBundle",
 		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "merged-support-bundle-spec",
+		Metadata: troubleshootv1beta2.SupportBundleMetadata{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "merged-support-bundle-spec",
+			},
+			RunHostCollectorsInPod: false,
 		},
 	}
+
+	var enableRunHostCollectorsInPod bool
+
 	for _, sb := range kinds.SupportBundlesV1Beta2 {
 		sb := sb
 		mainBundle = supportbundle.ConcatSpec(mainBundle, &sb)
+		enableRunHostCollectorsInPod = enableRunHostCollectorsInPod || sb.Metadata.RunHostCollectorsInPod
```

Comment:
Will this cause a panic if `sb.Metadata` was `nil` for some reason? Should we check that it isn't nil here and then just set `enableRunHostCollecorsInPod := sb.Metadata.RunHostCollectorsInPod`?

Reply:
yes, we should check that. I have added the checking part